src/OrleansAWSUtils/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+using OrleansAWSUtils.Configuration;
+
+namespace OrleansAWSUtils
+{
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Configure DI container with DynamoDBMembershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configureOptions"></param>
+        public static IServiceCollection UseDynamoDBMembershipTable(this IServiceCollection services,
+            Action<DynamoDBMembershipTableOptions> configureOptions)
+        {
+            services.Configure<DynamoDBMembershipTableOptions>(configureOptions);
+            services.AddSingleton<IMembershipTable, DynamoDBMembershipTable>();
+            return services;
+        }
+
+        /// <summary>
+        /// Configure DI container with DynamoDBMembershipTable
+        /// </summary>
+        public static IServiceCollection UseDynamoDBMembershipTable(this IServiceCollection services,
+            IConfiguration config)
+        {
+            services.Configure<DynamoDBMembershipTableOptions>(config);
+            services.AddSingleton<IMembershipTable, DynamoDBMembershipTable>();
+            return services;
+        }
+
+        /// <summary>
+        /// Configure DI container with DynamoDBMemebershipTable and get its configuration from <see cref="GlobalConfiguration"/>
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection UseDynamoDBMemebershipTableFromLegacyConfigurationSupport(
```

Comment:
@ReubenBond I assume this is a candidate to use the ApplicationPartManager instead of the user being explicit with both configuring the XML and then have to call this method?

Reply:
? mind explain a bit more on the context>? I'm not quite follow

Reply:
Hehe, I was summoning @ReubenBond , as this is related to stuff not yet checked in, but will be part of the type discovery part (instead of the full assembly load and scanning we are currently doing).
What I mean is that once that's in, we could have the method in the host builder that when you pass in the legacy configuration object, then it uses the same discovery process to find these types that convert the legacy configuration into strongly typed options, even if they reside on different provider specific assemblies. Not to worry about it for this PR, only after Reuben introduces that functionality.

Reply:
BTW, as we talked about in person, we do want to do something to avoid users having to call this method from each provider implementation explicitly, whether that's using the upcoming AssemblyPartsManager, or by using and approach similar to the one in MembershipTableFactory today (where you call `AssemblyLoader.LoadAndCreateInstance<ILegacyMembershipConfigurator>` or something like that)